lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/MultiRuntimeYarnBootstrapDriverConfigGenerator.java

```diff
@@ -0,0 +1,271 @@
+          final MultiRuntimeDefinitionBuilder builder) {
+    // create and serialize local configuration if defined
+    final Configuration localModule = LocalDriverConfiguration.CONF
+            .set(LocalDriverConfiguration.MAX_NUMBER_OF_EVALUATORS,
+                    localAppSubmissionParams.getMaxNumberOfConcurrentEvaluators())
+            // ROOT FOLDER will point to the current runtime directory
+            .set(LocalDriverConfiguration.ROOT_FOLDER, ".")
+            .set(LocalDriverConfiguration.JVM_HEAP_SLACK, 0.0)
+            .set(LocalDriverConfiguration.CLIENT_REMOTE_IDENTIFIER, ClientRemoteIdentifier.NONE)
+            .set(LocalDriverConfiguration.JOB_IDENTIFIER,
+                    jobSubmissionParameters.getJobId().toString())
+            .set(LocalDriverConfiguration.RUNTIME_NAMES,
+                    org.apache.reef.runtime.local.driver.RuntimeIdentifier.RUNTIME_NAME)
+            .build();
+
+    // add local runtime to the builder
+    builder.addRuntime(localModule, org.apache.reef.runtime.local.driver.RuntimeIdentifier.RUNTIME_NAME);
+  }
+
+  public String writeDriverConfigurationFile(final String bootstrapJobArgsLocation,
+                                             final String bootstrapAppArgsLocation) throws IOException {
+    final File bootstrapJobArgsFile = new File(bootstrapJobArgsLocation).getCanonicalFile();
+    final File bootstrapAppArgsFile = new File(bootstrapAppArgsLocation);
+
+    final AvroYarnJobSubmissionParameters yarnBootstrapJobArgs =
+        readYarnJobSubmissionParametersFromFile(bootstrapJobArgsFile);
+
+    final AvroMultiRuntimeAppSubmissionParameters multiruntimeBootstrapAppArgs =
+        readMultiRuntimeAppSubmissionParametersFromFile(bootstrapAppArgsFile);
+
+    final String driverConfigPath = reefFileNames.getDriverConfigurationPath();
+
+    this.configurationSerializer.toFile(
+            getMultiRuntimeDriverConfiguration(
+                    yarnBootstrapJobArgs, multiruntimeBootstrapAppArgs),
+        new File(driverConfigPath));
+
+    return driverConfigPath;
+  }
+
+  Configuration getMultiRuntimeDriverConfiguration(
+          final AvroYarnJobSubmissionParameters yarnJobSubmissionParams,
+          final AvroMultiRuntimeAppSubmissionParameters multiruntimeAppSubmissionParams) {
+
+    // read yarn job submission parameters
+    final AvroJobSubmissionParameters jobSubmissionParameters =
+        yarnJobSubmissionParams.getSharedJobSubmissionParameters();
+
+    // generate multi runtime definition
+    final MultiRuntimeDefinitionBuilder multiRuntimeDefinitionBuilder = new MultiRuntimeDefinitionBuilder();
+
+    if (multiruntimeAppSubmissionParams.getLocalRuntimeAppParameters() != null){
+      addLocalRuntimeDefinition(
+              multiruntimeAppSubmissionParams.getLocalRuntimeAppParameters(),
+              jobSubmissionParameters, multiRuntimeDefinitionBuilder);
+    }
+
+    if (multiruntimeAppSubmissionParams.getYarnRuntimeAppParameters() != null){
```

Comment:
What is bound to REEF runtime is the multiruntime:
public static final ConfigurationModule CONF = new MultiRuntimeDriverConfiguration()
      .bindImplementation(ResourceLaunchHandler.class, MultiRuntimeResourceLaunchHandler.class)
      .bindImplementation(ResourceRequestHandler.class, MultiRuntimeResourceRequestHandler.class)
      .bindImplementation(ResourceReleaseHandler.class, MultiRuntimeResourceReleaseHandler.class)
      .bindImplementation(ResourceManagerStartHandler.class, MultiRuntimeResourceManagerStartHandler.class)
      .bindImplementation(ResourceManagerStopHandler.class, MultiRuntimeResourceManagerStopHandler.class)
      .bindNamedParameter(SerializedRuntimeDefinition.class, SERIALIZED_RUNTIME_DEFINITION)
      .bindNamedParameter(LaunchID.class, JOB_IDENTIFIER)
      .bindNamedParameter(JobIdentifier.class, JOB_IDENTIFIER)
      .bindNamedParameter(ClientRemoteIdentifier.class, CLIENT_REMOTE_IDENTIFIER)
      .bindSetEntry(DefinedRuntimes.class, RUNTIME_NAMES)

I don't think we need to launch any other YArn handlers if we are not actually requesting Yarn containers.

I do bind the following:

        .bindNamedParameter(JobSubmissionDirectoryPrefix.class,
            yarnJobSubmissionParams.getJobSubmissionDirectoryPrefix().toString())
        .bindImplementation(RuntimeClasspathProvider.class, YarnClasspathProvider.class)
        .bindConstructor(YarnConfiguration.class, YarnConfigurationConstructor.class)

Yarn job submission parameters are read in as a JObParams not App params. I will test it again before merge on YArRest/HDI